tests/devices/qubit/test_sampling.py

```diff
@@ -252,3 +252,99 @@ def test_approximate_expval_measure(self):
 
         assert result != 0
         assert np.allclose(result, 0, atol=0.03)
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
+        [
+            [(100,), 1],
+            [((100, 1),), 1],
+            [((100, 2),), 2],
+            [(100, 100), 2],
+            [(100, 200), 2],
+            [(100, 100, 200), 3],
+            [(200, (100, 2)), 3],
+        ],
+    )
+    def test_sample_measure_shot_vector(self, shots, total_copies):
+        """Test that a sample measurement with shot vectors works as expected"""
+        state = qml.math.array(two_qubit_state)
+        shots = qml.measurements.Shots(shots)
+        mp = qml.sample(wires=range(2))
+
+        result = measure_with_samples(mp, state, shots=shots)
+
+        if total_copies == 1:
+            assert isinstance(result, np.ndarray)
+            result = (result,)
+
+        assert isinstance(result, tuple)
+        assert len(result) == total_copies
+
+        for res, sh in zip(result, shots):
+            assert res.shape == (sh, 2)
+            assert res.dtype == np.bool8
+            assert all(qml.math.allequal(s, [0, 1]) or qml.math.allequal(s, [1, 0]) for s in res)
+
+    @pytest.mark.parametrize(
+        "shots, total_copies",
+        [
+            [(100,), 1],
+            [((100, 1),), 1],
+            [((100, 2),), 2],
+            [(100, 100), 2],
+            [(100, 200), 2],
+            [(100, 100, 200), 3],
+            [(200, (100, 2)), 3],
+        ],
+    )
+    def test_prob_measure_shot_vector(self, shots, total_copies):
+        """Test that a probability measurement with shot vectors works as expected"""
+        state = qml.math.array(two_qubit_state)
+        shots = qml.measurements.Shots(shots)
+        mp = qml.probs(wires=range(2))
```

Comment:
We need tests for non-integer wire labels. The device maps user defined wire labels into `[0, .., n-1]` for an n-wire device. Currently, if `mp.wires == Wires(["a", "b"])`, the code likely won't work, so you need to map the user wires to device wires. See [here](https://github.com/PennyLaneAI/pennylane/blob/279360e5d04201a2c6a03b4b6d83c3c34c8af020/pennylane/devices/qubit/simulate.py#L49)

Reply:
I don't think this is a case we have to worry about, since `measure_with_samples` is meant to only be called from `simulate` anyways, and `simulate` already maps all the measurement wires to be integers. Of course this is still something to bare in mind when integrating with `simulate`.

Reply:
@eddddddy is correct.  This will be like `apply_operation`, in that it assumes that all the wires have been mapped to positive integers that correspond to dimensions of the state.

Maybe we should explicitly write out this assumption in the function docstring.

Reply:
Will add this to the docstring 👍

Reply:
Forgot about that. Thanks!

Reply:
Forgot about the wire mapping done in `simulate`. Thanks!